src/cbc_sdk/workload/compliance_assessment.py

```diff
@@ -0,0 +1,465 @@
+            ...     print(section.section_name, section.section_id)
+        """
+        url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/sections"
+        results = self._cb.get_object(url)
+        for item in results:
+            yield ComplianceBenchmark(self._cb, initial_data=item)
+
+    def execute_action(self, action, device_ids=None):
+        """
+        Execute a specified action on devices within on a Benchmark Set, or specific devives
+        within a Benchmark Set only.
+
+        Args:
+            action (str): The action to be executed. Available:
+                'ENABLE': Enable the object.
+                'DISABLE': Disable the object.
+                'REASSESS': Reassess the object.
+
+            device_ids (str or list, optional): IDs of devices on which the action will be executed.
+                If specified as a string, only one device will be targeted. If specified as a list,
+                the action will be executed on multiple devices. Default is None.
+
+        Returns:
+            dict: JSON response containing information about the executed action.
+
+        Raises:
+            ApiError: If the provided action is not one of the allowed actions.
+
+        Example:
+            To reassess an object:
+            benchmark_sets = cb.select(ComplianceBenchmark)
+            benchmark_sets[0].execute_action('REASSESS')
+        """
+        ACTIONS = ('ENABLE', 'DISABLE', 'REASSESS')
+
+        if action.upper() not in ACTIONS:
+            message = (
+                f"\nAction type is required."
+                f"\nAvailable action types: {', '.join(ACTIONS)}"
+                f"\nExample:\nbenchmark_sets = cb.select(ComplianceBenchmark)"
+                f"\nbenchmark_sets[0].execute_action('REASSESS')"
+            )
+            raise ApiError(message)
+
+        args = {"action": action}
+        if device_ids:
+            args['device_ids'] = [device_ids] if isinstance(device_ids, str) else device_ids
+
+        url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/actions"
+        return self._cb.post_object(url, body=args).json()
+
+    def get_benchmark_set_summary(self):
+        """
+        Fetches the compliance summary for the current benchmark set.
+
+        This method constructs the URL for the compliance summary of the benchmark set associated with the current instance,
+        fetches the summary data using the Carbon Black API, and yields Compliance objects for each item in the summary.
+
+        Returns:
+            generator of ComplianceBenchmark: A generator yielding Compliance objects, each representing a summary item.
+        """
+        url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/compliance/summary"
+        results = self._cb.get_object(url)
+        for item in results:
+            yield ComplianceBenchmark(self._cb, initial_data=item)
+
```

Comment:
We need to look into adding the following endpoints but I think we can enable them as single functions inside ComplianceBenchmark as long as we expose the query, sort and criteria and return the raw json response
- [Search Device Compliance Summaries](https://developer.carbonblack.com/reference/carbon-black-cloud/workload-protection/latest/cis-benchmark-api/#search-device-compliance-summaries)
- [Search Rule Compliance Summaries](https://developer.carbonblack.com/reference/carbon-black-cloud/workload-protection/latest/cis-benchmark-api/#search-rule-compliance-summaries)
- [Search Rule Compliance Results for a Device](https://developer.carbonblack.com/reference/carbon-black-cloud/workload-protection/latest/cis-benchmark-api/#search-rule-compliance-results-for-a-device)
- [Search Rule Results for Devices](https://developer.carbonblack.com/reference/carbon-black-cloud/workload-protection/latest/cis-benchmark-api/#search-rule-results-for-devices)